src/components/RestaurantList/index.jsx

```diff
@@ -1,24 +1,21 @@
 import React, { useEffect, useState } from 'react';
-import axios from 'axios';
 import RestaurantCard from '../RestaurantCard';
-const apiUrl = process.env.REACT_APP_BACKEND_URL;
+import { getRestaurantsByCuisine } from "../../services/restaurantService"; // Import the service
+
+
 const RestaurantList = ({ selectedCuisine }) => {
   const [restaurants, setRestaurants] = useState([]);
 
   useEffect(() => {
     const fetchData = async () => {
       try {
-        const url = `${apiUrl}/api/restaurants/getRestaurantsByCuisine`;
-        const data = {"cuisine":selectedCuisine}
-        const result = await axios.post(url, data);
-        console.log("result,",result.data)
-        setRestaurants(result.data)
+        const data = await getRestaurantsByCuisine(selectedCuisine);
+        setRestaurants(data);
       } catch (error) {
-        // Handle errors, e.g., display an error message
-        console.error('Error:', error);
+        console.error("Error:", error);
       }
     };
-  
+
     fetchData();
   }, [selectedCuisine]);
```

Comment:
This would be a great example to move into a hook called `useFetchRestaurants`
```js
// src/components/RestaurantList/index.jsx
const RestaurantList = ({ selectedCuisine }) => {
  const { restaurants, isLoading, isError } = useFetchRestaurants({ selectedCuisine })
  if (isLoading) {
    return (
    <div>
      <h1>Restaurants Near You</h1>
      <div>Loading...</div>
    </div>
  )
  if (isError) {
    return (
    <div>
      <h1>Restaurants Near You</h1>
      <div>Something went wrong</div>
    </div>
  )
  return (
    <div>
      <h1>Restaurants Near You</h1>
      {restaurants.map((restaurant) => (
        <div key={restaurant._id}>
        <RestaurantCard cardDetails={restaurant}/>
        </div>
      ))}
    </div>
  );
};
```

```js
// src/components/RestaurantList/hooks/useFetchRestaurants.jsx
const useFetchRestaurants = ({ selectedCuisine }) => {
  const [restaurants, setRestaurants] = useState([]);
  const [isLoading, setIsLoading] = useState(false);
  const [error, setError] = useState();

  useEffect(() => {
    const fetchData = async () => {
      setIsLoading(true)
      setError(undefined)
      try {
        const url = `${apiUrl}/api/restaurants/getRestaurantsByCuisine`;
        const data = {"cuisine":selectedCuisine}
        const result = await axios.post(url, data);
        console.log("result,",result.data)
        setRestaurants(result.data)
        const data = await getRestaurantsByCuisine(selectedCuisine);
        setRestaurants(data);
      } catch (error) {
        // Handle errors, e.g., display an error message
        console.error('Error:', error);
        console.error("Error:", error);
        setError(error)
      } finally {
        setIsLoading(false)
      }
    };
    fetchData();
  }, [selectedCuisine, setIsLoading, setError]);
  return {
    restaurants,
    isLoading,
    isError,
    error,
  }
}
```

Reply:
Hi David,
we have implemented the hooks which contains business logic and separated it from presentation layer